File: Arrays&Strings/SlidingWindow/findMaxAverage.js
function findMaxAverage(nums, k) {
    let curr = 0;
    let answer = 0;

    for (let i = 0; i < k; i++) {
        curr += nums[i];
    }

    answer = curr / k;

    for (let i = k; i < nums.length; i++) {
        curr += nums[i];
        curr -= nums[i - k];
        answer = Math.max(answer, curr / k);
    }

    return answer;
}

console.log(findMaxAverage([1,12,-5,-6,50,3], 4))

File: Arrays&Strings/SlidingWindow/findMaxAverage.py
def findMaxAverage(nums: list[int], k: int) -> int:
    curr, answer = 0, 0

    for i in range(k):
        curr += nums[i]

    answer = curr / k

    for i in range(k, len(nums)):
        curr += nums[i]
        curr -= nums[i - k]
        answer = max(answer, curr / k)

    return answer

print(findMaxAverage([1,12,-5,-6,50,3], 4))